src/main/java/com/google/devtools/build/lib/actions/Spawns.java

```diff
@@ -152,21 +155,72 @@ public static Duration getTimeout(Spawn spawn) throws ExecException {
    * defaultTimeout, or {@code Duration.ZERO} if that is null.
    */
   public static Duration getTimeout(Spawn spawn, Duration defaultTimeout) throws ExecException {
-    String timeoutStr = spawn.getExecutionInfo().get(ExecutionRequirements.TIMEOUT);
-    if (timeoutStr == null) {
-      return defaultTimeout == null ? Duration.ZERO : defaultTimeout;
+    Duration timeout = null;
+
+    for (Map.Entry<String, String> entry: spawn.getExecutionInfo().entrySet()) {
+        String value = getTimeoutValueFromExecutionInfoKeyOrValue(spawn,entry);
+        if (value != null) {
+            if (timeout != null) {
+                throw new UserExecException(
+                    FailureDetail.newBuilder()
+                        .setMessage("multiple timeout values specified")
+                        .setSpawn(FailureDetails.Spawn.newBuilder().setCode(Code.DUPLICATE_TIMEOUT_TAGS))
```

Comment:
I'm not sure that this is the behavior we want on duplicates. If the rule author says `execution_requirements = {"timeout": "5m"}` but the user says `my_library(tags = ["timeout:10m"])`, wouldn't we want the latter to override the former?

It might be acceptable to always take the more permissive option; otherwise we'd have to track the provenance of execution requirements, which might be hard.